packages/backend/src/open_payments/payment/incoming/model.ts

```diff
@@ -239,39 +249,56 @@ export class IncomingPayment
     return payment
   }
 
-  public toOpenPaymentsType({
-    ilpStreamConnection
-  }: {
+  public async toOpenPaymentsType(): Promise<OpenPaymentsIncomingPayment>
+  public async toOpenPaymentsType(
     ilpStreamConnection: Connection
-  }): OpenPaymentsIncomingPayment {
-    return {
-      id: this.url,
-      paymentPointer: this.paymentPointer.url,
+  ): Promise<OpenPaymentsIncomingPaymentWithConnection>
+  public async toOpenPaymentsType(
+    ilpStreamConnection: string
+  ): Promise<OpenPaymentsIncomingPaymentWithConnectionUrl>
+  public async toOpenPaymentsType(
+    ilpStreamConnection?: Connection | string
+  ): Promise<
+    | OpenPaymentsIncomingPaymentWithConnection
+    | OpenPaymentsIncomingPaymentWithConnectionUrl
+  >
+  public async toOpenPaymentsType(
+    ilpStreamConnection?: Connection | string
+  ): Promise<
+    | OpenPaymentsIncomingPayment
+    | OpenPaymentsIncomingPaymentWithConnection
+    | OpenPaymentsIncomingPaymentWithConnectionUrl
+  > {
+    const paymentPointer = await this.getPaymentPointer()
+    const baseIncomingPayment: OpenPaymentsIncomingPayment = {
+      id: await this.getUrl(paymentPointer),
+      paymentPointer: paymentPointer.url,
       incomingAmount: this.incomingAmount
         ? serializeAmount(this.incomingAmount)
         : undefined,
       receivedAmount: serializeAmount(this.receivedAmount),
       completed: this.completed,
+      description: this.description ?? undefined,
+      externalRef: this.externalRef ?? undefined,
       createdAt: this.createdAt.toISOString(),
       updatedAt: this.updatedAt.toISOString(),
-      expiresAt: this.expiresAt.toISOString(),
+      expiresAt: this.expiresAt.toISOString()
+    }
+
+    if (!ilpStreamConnection) {
+      return baseIncomingPayment
+    }
+
+    if (typeof ilpStreamConnection === 'string') {
+      return {
+        ...baseIncomingPayment,
+        ilpStreamConnection
+      }
+    }
+
+    return {
+      ...baseIncomingPayment,
       ilpStreamConnection: ilpStreamConnection.toOpenPaymentsType()
     }
   }
 }
-
-// TODO: disallow undefined
-// https://github.com/interledger/rafiki/issues/594
-export type IncomingPaymentJSON = {
-  id: string
-  paymentPointer: string
-  incomingAmount?: AmountJSON
-  receivedAmount: AmountJSON
```

Comment:
@wilsonianb Keeping `AmountJSON` over `OpenPaymentsAmount` makes sense right? The Amount interface is a more generic thing we use across the code, not necessarily being an "open payments" concept